### PR TITLE
[Luoweichao] support logging via EFK stack

### PR DESCRIPTION
**Pull request by chaosreload**
1. add es-domain module to setup OpenSearch service
2. update user_data.sh to setup filebeat and rsyslogd

### CODE REVIEW

**Comment by chaosreload** [comment]
Just updated:
1. setup more shards for the index to spread the requests evenly to the data nodes
2. process json inside filebeat
3. remove the problematic 'aws_subnet_ids'

File: infra-terraform/instance/single-region-with-efk/providers.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }
+
+  required_version = "~> 1.0"
+
+  backend "s3" {
+    bucket = "terraform-nvstress-test-sd"
+    key    = "luoweichao-stl"
+    region = "us-west-2"
+    profile = "aws_aws-cube-nvstressclient-bd-us_app.luoweichao"
```
Comment (jxie-sd) [inline infra-terraform/instance/single-region-with-efk/providers.tf:23]: we can't not have this profile in the source code

File: infra-terraform/instance/single-region-with-efk/providers.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+  profile = "aws_aws-cube-nvstressclient-bd-us_app.luoweichao"
```
Comment (jxie-sd) [inline infra-terraform/instance/single-region-with-efk/providers.tf:29]: we can't not have this profile in the source code

**Review by jxie-sd — CHANGES_REQUESTED** [review]

**Comment by jxie-sd** [comment]
I'm looking at this PR. I don't know if we can just merge [infra-terraform/instance/single-region-with-efk ](https://github.com/seconddinner/stress-test-loader/pull/7/files#diff-2ed3c9a2ac2a7aaede3015cb81e17536beefc50c96b1c2e01d2a4082fc8f6481) into [infra-terraform/instance/single-region ](https://github.com/seconddinner/stress-test-loader/pull/7/files#diff-2ed3c9a2ac2a7aaede3015cb81e17536beefc50c96b1c2e01d2a4082fc8f6481) and make ES a variable, and if it's turned on, add ES.

what is the use of infra-terraform/opensearch-domain

File: infra-terraform/terraform-modules/iam/iam.tf [hunk 3]
```diff
@@ -33,16 +37,33 @@ resource "aws_iam_role_policy" "stress_test_client_read" {
...
+                "s3:GetObject"
+            ],
+            "Effect": "Allow",
+            "Resource": "arn:aws:s3:::cubestressclientartifactbucket-nv/*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "secretsmanager:GetResourcePolicy",
+                "secretsmanager:GetSecretValue",
```
Comment (jxie-sd) [inline infra-terraform/terraform-modules/iam/iam.tf:53]: why does the terraform EC2 need to read secrets? can we limit to smaller subset of the secret?

Comment (jxie-sd) [reply infra-terraform/terraform-modules/iam/iam.tf:53]: do something like following is a lot safer

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "secretsmanager:GetResourcePolicy",
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
                "secretsmanager:ListSecretVersionIds"
            ],
            "Resource": [
                "arn:aws:secretsmanager:us-west-2:111122223333:secret:aes128-1a2b3c",
                "arn:aws:secretsmanager:us-west-2:111122223333:secret:aes192-4D5e6F",
                "arn:aws:secretsmanager:us-west-2:111122223333:secret:aes256-7g8H9i"
            ]
        },
        {
            "Effect": "Allow",
            "Action": "secretsmanager:ListSecrets",
            "Resource": "*"
        }
    ]
}

**Review by jxie-sd — CHANGES_REQUESTED** [review]

**Comment by jxie-sd** [comment]
you need to rebase on master to resolve conflict

**Comment by jxie-sd** [comment]
-var 'create_esdomain=false' will produce: 

│ Error: your query returned no results
│ 
│   with data.aws_elasticsearch_domain.es[0],
│   on es-domain.tf line 15, in data "aws_elasticsearch_domain" "es":
│   15: data "aws_elasticsearch_domain" "es" {
│

File: infra-terraform/terraform-modules/iam/iam.tf (hunk 3 above)
Comment (chaosreload) [reply infra-terraform/terraform-modules/iam/iam.tf:53]: I wanted to store the password of Kibana in the secretsmanager, so I added those related policies. I'll limit the affected resources only to the password

**Comment by chaosreload** [comment]
> 

I follow these steps to create the PR:
1. edit codes in forked luoweichao branch
2. commit and push to origin
3. create PR from 'luoweichao' branch to main branch in upstream

Could you share the operation to rebase on main branch?

**Comment by chaosreload** [comment]
> │ Error: your query returned no results

It fails when you specify  'create_esdomain=false' and a non-exist ElasticSearch domain. You can specify an existing ElasticSearch domain in the 'esdomain' variable or create a new one by specifying 'create_esdomain=true'.

File: infra-terraform/terraform-modules/iam/iam.tf (hunk 3 above)
Comment (jxie-sd) [reply infra-terraform/terraform-modules/iam/iam.tf:53]: got it. make sense. Let limit only for certain resources that Kibana needs

**Comment by jxie-sd** [comment]
@chaosreload Like what we discussed, we need a control variable that take out ES domains, so it doesn't even build the ES in sometimes.

**Review by jxie-sd — APPROVED** [review]
approve this for now. 

We still need to work on create_esdomain = false part

**Pull request merged**